### PR TITLE
Change Replicated distribution to a record

### PR DESCRIPTION
**Pull request by bradcray**
This follows in the path of #22951 and #22657 by converting Replicated into a record, like Block and Cyclic.  The approach taken was:

* create a `Replicated` record, similar to `Block` and `Cyclic`.  `Replicated` is simpler because it's not generic
* change the old `Replicated` class to `ReplicatedImpl`
* be sure to add a `.dsiGetDist()` method to `ReplicatedDom`
* update tests to stop wrapping `Replicated` types/values with `dmap`
* merge by two `dmap` deprecation tests into one, and extend it to check `Replicated`

While here, I also did some (unnecessary) cleanup to tests that are in our directory but not compiled, because a `git grep` caused me to find some `dmap`s that I wanted to update for consistency, so then I made other updates to check my work.

I also wanted to note that it looks like `Replicated` takes a string in its initializer which is unused, and if I were working more aggressively at stabilizing `Replicated` rather than just retiring `dmap`, I would probably have removed it, but that felt like a step too far for this PR.

### CODE REVIEW

File: test/parallel/forall/reduce-intents/ri-4-arrdom.chpl [hunk 1]
```diff
@@ -25,13 +25,13 @@ proc setupDistributions() {
     return new Block(rank=2, boundingBox={1..d, 1..d});
 
   else if distType == DistType.cyclic then
-    return new dmap(new Cyclic(startIdx=(0,0)));
```
Comment (jeremiah-corrado) [inline test/parallel/forall/reduce-intents/ri-4-arrdom.chpl:28]: Had this not been emitting a deprecation warning before?

File: test/studies/hpcc/HPL/vass/hpl.hpcc2012.chpl [hunk 2]
```diff
@@ -13,7 +13,7 @@ use LinearAlgebra, Random, Time;
 // Use the distributions we need for this computation
 //
 use DimensionalDist2D, ReplicatedDim, BlockCycDim;
-use ReplicatedDist;
+use ReplicatedDist, ChplConfig;
```
Comment (jeremiah-corrado) [inline test/studies/hpcc/HPL/vass/hpl.hpcc2012.chpl:16]: I'm not seeing why this was imported here. Had `ReplicatedDist` previously been importing `ChplConfig` with it?

**Review by jeremiah-corrado — APPROVED** [review]
Looks good to me! Just a couple of questions:

File: test/parallel/forall/reduce-intents/ri-4-arrdom.chpl (hunk 1 above)
Comment (bradcray) [reply test/parallel/forall/reduce-intents/ri-4-arrdom.chpl:28]: This test (and the next) weren't actually being tested.  I was just doing some git grepping of dmap + Cyclic | Replicated and ended up updating these tests despite their not being tested (and then kept going and updating other aspects of them that trivially weren't compiling).

**Pull request merged**